src/semaphore.rs

```diff
@@ -171,8 +272,35 @@ impl Semaphore {
         }
     }
 
-    /// Acquire permits from the semaphore, potentially unfairly stealing permits from a task that
-    /// is waiting for permits.
+    /// Acquire permits from the semaphore, potentially unfairly stealing
+    /// permits from a task that is waiting for permits.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use std::future::Future;
+    ///
+    /// use unsync::Semaphore;
+    ///
+    /// # #[tokio::main(flavor = "current_thread")] async fn main() {
+    /// # unsync::utils::noop_cx!(cx);
+    /// let semaphore = Semaphore::new(1);
+    ///
+    /// let future = semaphore.acquire(2);
+    /// tokio::pin!(future);
+    /// assert!(future.as_mut().poll(cx).is_pending());
+    ///
+    /// assert!(semaphore.try_acquire(1).is_none());
+    /// // steal one permit from `future` which is in the process of acquiring permits.
+    /// let permit = semaphore.acquire_unfair(1).await;
+    /// drop(permit);
+    ///
+    /// drop(future);
+    ///
+    /// // TODO: why doesn't this work?
```

Comment:
@SabrinaJewson is this the drop issue you mentioned, am I misunderstanding how the API should work, or something else?

Reply:
Sorry for the noise, this was just a scope issue 🙃